src/redisearch_rs/trie_bencher/src/bencher.rs

```diff
@@ -0,0 +1,337 @@
+use std::{
+    ffi::{CString, c_char, c_void},
+    hint::black_box,
+    ptr::NonNull,
+};
+
+/// A helper struct for benchmarking operations on different trie map implementations.
+pub struct OperationBencher<'a> {
+    c: &'a mut Criterion,
+    rust_map: RustTrieMap,
+    radix: RustRadixTrie,
+    c_map: *mut CTrieMap,
+}
+
+impl<'a> OperationBencher<'a> {
+    /// Initialize a new bencher by creating a new instance of each trie map implementation
+    /// for the same corpus of words.
+    pub fn new(c: &'a mut Criterion, contents: String) -> Self {
+        let c_char_words = contents
+            .split_whitespace()
+            .map(str2c_char)
+            .collect::<Vec<_>>();
+        let rust_map = rust_load(&c_char_words);
+
+        let u8_words = contents
+            .split_whitespace()
+            .map(crate::str2u8)
+            .collect::<Vec<_>>();
+        let radix = radix_load(u8_words);
+
+        let c_words = contents
+            .split_whitespace()
+            .map(|word| CString::new(word).expect("CString conversion failed"))
+            .map(|s| {
+                let len: u16 = s.as_bytes_with_nul().len().try_into().unwrap();
+                let converted = s.into_raw();
+                (len, converted)
+            })
+            .collect::<Vec<_>>();
+        let c_map = c_load(c_words);
+
+        OperationBencher {
+            c,
+            rust_map,
+            radix,
+            c_map,
+        }
+    }
+
+    /// Benchmark the find operation.
+    ///
+    /// The benchmark group will be marked with the given label.
+    pub fn find_group(&mut self, word: &str, label: &str) {
+        let mut group = self.c.benchmark_group(label);
+        find_rust_benchmark(&mut group, self.rust_map.clone(), word);
+        find_radix_benchmark(&mut group, self.radix.clone(), word);
+        find_c_benchmark(&mut group, self.c_map, word);
+        group.finish();
+    }
+
+    /// Benchmark the insert operation.
+    ///
+    /// The benchmark group will be marked with the given label.
+    pub fn insert_group(&mut self, word: &str, label: &str) {
+        let mut group = self.c.benchmark_group(label);
+        insert_rust_benchmark(&mut group, self.rust_map.clone(), word);
+        insert_radix_benchmark(&mut group, self.radix.clone(), word);
+        insert_c_benchmark(&mut group, self.c_map, word);
+        group.finish();
+    }
+
+    /// Benchmark the removal operation.
+    ///
+    /// The benchmark group will be marked with the given label.
+    pub fn remove_group(&mut self, word: &str, label: &str) {
+        let mut group = self.c.benchmark_group(label);
+        remove_rust_benchmark(&mut group, self.rust_map.clone(), word);
+        remove_radix_benchmark(&mut group, self.radix.clone(), word);
+        remove_c_benchmark(&mut group, self.c_map, word);
+        group.finish();
+    }
+
+    /// Benchmark loading a corpus of words.
+    pub fn load_group(&mut self, contents: String) {
```

Comment:
can we keep file format constant? 
`insert_group` for example calling specific benchmark functions, but `load_group` inlines them